libcontainer/cgroups/fs/cpuset_test.go

```diff
@@ -65,3 +184,68 @@ func TestCpusetSetMems(t *testing.T) {
 		t.Fatal("Got the wrong value, set cpuset.mems failed.")
 	}
 }
+
+func TestCpusetStatsCorrect(t *testing.T) {
+	helper := NewCgroupTestUtil("cpuset", t)
+	defer helper.cleanup()
+	helper.writeFileContents(cpusetTestFiles)
+
+	cpuset := &CpusetGroup{}
+	actualStats := *cgroups.NewStats()
+	err := cpuset.GetStats(helper.CgroupPath, &actualStats)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectedStats := cgroups.CpusetStats{
+		Cpus:                  []uint16{0, 1, 2, 7, 12, 13, 14},
+		CpuExclusive:          1,
+		Mems:                  []uint16{1, 2, 3, 4, 6, 9},
+		MemoryMigrate:         1,
+		MemHardwall:           0,
+		MemExclusive:          0,
+		MemorySpreadPage:      0,
+		MemorySpreadSlab:      1,
+		MemoryPressure:        34377,
+		SchedLoadBalance:      1,
+		SchedRelaxDomainLevel: -1}
+	if !reflect.DeepEqual(expectedStats, actualStats.CpusetStats) {
+		t.Fatalf("Expected Cpuset stats usage %#v but found %#v\n",
+			expectedStats, actualStats.CpusetStats)
+	}
+
+}
+
+func TestCpusetStatsMissingFiles(t *testing.T) {
+	for _, failCase := range cpusetStatFailureTestCases {
+		helper := NewCgroupTestUtil("cpuset", t)
+		defer helper.cleanup()
+
+		tempCpusetTestFiles := map[string]string{}
+		for i, v := range cpusetTestFiles {
+			tempCpusetTestFiles[i] = v
+		}
+
+		if failCase.removeFile {
+			delete(tempCpusetTestFiles, failCase.filename)
+			helper.writeFileContents(tempCpusetTestFiles)
+			cpuset := &CpusetGroup{}
+			actualStats := *cgroups.NewStats()
+			err := cpuset.GetStats(helper.CgroupPath, &actualStats)
+
+			if err != nil {
+				t.Errorf(fmt.Sprintf("test case '%s' failed unexpectedly: '%s'", failCase.desc, err))
```

Comment:
nit: no need to use `fmt.Sprintf` here, as `t.Errorf` is perfectly capable of handling `%`-style arguments.

Reply:
Perhaps use sub-tests for these cases, then it would not be needed to include the name of the failed test in the error?